src/impl_ptrs.rs

```diff
@@ -0,0 +1,172 @@
+use dimension::stride_offset;
+use imp_prelude::*;
+use StrideShape;
+
+impl<A, D> ArrayPtr<A, D>
+where
+    D: Dimension,
+{
+    /// Create a new `ArrayPtr`.
```

Comment:
What do you think about names `RawArrayView` and `RawArrayViewMut` ?

Reply:
You mean instead of `ArrayPtr` and `ArrayPtrMut`? Those names are longer, but I think I prefer them because they carry less of the `*const T`/`*mut T` connotation. The names `ArrayPtr`/`ArrayPtrMut` could be confusing because you can't create an `ArrayPtr`/`ArrayPtrMut` given an arbitrary `*const T`/`*mut T`.

Another thing to consider is how this relates to #543, in particular the "non-null" constraint. Do we want to require the `ptr` in `ArrayPtr`/`ArrayPtrMut`/`RawArrayView`/`RawArrayViewMut` to be non-null? If yes, `RawArrayView` is nice because the name doesn't carry the connotation that the pointer could be null.

Reply:
I agree with your ideas about non-null over in that issue. We don't have a pressing need for using `NonNull<T>`.

I don't think using `NonNull` alone will solve anything about the reborrow methods though. I believe in my explanation in #412, which is not related to it being a const or mut pointer but due to the associated type `S::Data` making `S` fixed.  That said, Rust might have gotten more flexible while I was away..

Reply:
> I agree with your ideas about non-null over in that issue. But we also don't have a pressing need for using `NonNull<T>`.

What's your opinion about requiring non-nullness for the raw pointer types? I lean towards requiring non-null for the time being (but not guarantee that pointers returned from methods like `.as_ptr()` are non-null). We can always relax this constraint later.

> I don't think using `NonNull` alone will solve anything about the reborrow methods though. I believe in my explanation in #412, which is not related to it being a const or mut pointer but due to using the associated type `S::Elem` holds `S` fixed.

I just [tried it](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=62fb90d1198ed563095f0227cebd1db5), and you're right. `NonNull` doesn't solve the issue, but adding an `A` type parameter to `ArrayBase` does.

Reply:
yeah require nonull

Reply:
Okay, I'll add the non-null requirement to the docs (and anything else I might have missed from the constraints added in #543), and I'll rename `ArrayPtr`/`ArrayPtrMut` to `RawArrayView`/`RawArrayViewMut`.

Reply:
Done.